app/src/main/java/com/humberto/tasky/core/database/data_source/RoomLocalAgendaDataSource.kt

```diff
@@ -138,8 +144,14 @@ class RoomLocalAgendaDataSource(
     }
 
     override suspend fun deleteAllAgenda() {
-        taskDao.deleteAllTasks()
-        eventDao.deleteAllEvents()
-        reminderDao.deleteAllReminders()
+        withContext(Dispatchers.IO) {
+            agendaDatabase.withTransaction {
+                listOf(
+                    async { eventDao.deleteAllEvents() },
+                    async { reminderDao.deleteAllReminders() },
+                    async { taskDao.deleteAllTasks() }
+                )
```

Comment:
Don’t forget to call `.awaitAll()` so this function can be suspended until all the delete operations are complete. The caller of this function will use the completion of the function as an indicator that the delete operations have completed, and when you return early, that would be a misleading assumption.